src/pl/ogladajanime/src/eu/kanade/tachiyomi/animeextension/pl/ogladajanime/OgladajAnime.kt

```diff
@@ -0,0 +1,319 @@
+
+            if (check.isNotEmpty()) {
+                episode.name = if (episodeText.isNotEmpty()) {
+                    "[${episodeNumber.toInt()}] $episodeText ($episodeImg)"
+                } else {
+                    "Episode ${episodeNumber.toInt()} ($episodeImg)"
+                }
+                episode.episode_number = episodeNumber
+                episode.url = element.attr("ep_id")
+                episodeList.add(episode)
+            }
+        }
+        return episodeList.reversed()
+    }
+
+    override fun episodeListSelector(): String = throw UnsupportedOperationException()
+
+    override fun episodeFromElement(element: Element): SEpisode = throw UnsupportedOperationException()
+
+    // ============================ Video Links =============================
+
+    private fun getPlayerUrl(player: String): String {
+        val body = FormBody.Builder()
+            .add("action", "change_player_url")
+            .add("id", player)
+            .build()
+        return client.newCall(POST("$baseUrl/manager.php", apiHeaders, body))
+            .execute()
+            .use { response ->
+                response.body.string()
+                    .substringAfter("\"data\":\"")
+                    .substringBefore("\",")
+                    .replace("\\", "")
+            }
+    }
+    override suspend fun getVideoList(episode: SEpisode): List<Video> {
+        val body = FormBody.Builder()
+            .add("action", "get_player_list")
+            .add("id", episode.url)
+            .build()
+        val response = client.newCall(POST("$baseUrl/manager.php", apiHeaders, body)).await()
+
+        val serverList = mutableListOf<Pair<String, String>>()
+
+        val jsonObject = JSONObject(response.body.string())
+        val dataString = jsonObject.getString("data")
+
+        val dataObject = JSONObject(dataString)
+        val playersArray = dataObject.getJSONArray("players")
+
+        (0 until playersArray.length()).forEach {
+            val player = playersArray.getJSONObject(it)
+            val id = player.getString("id")
+            var sub = player.getString("sub").uppercase()
+            val ismy = player.getInt("ismy")
+
+            // Modify sub if it's PL
+            if (player.getString("audio") == "pl") {
+                sub = "Dub PL"
+            }
+
+            val prefix = if (ismy > 0) {
+                "[$sub/Odwrócone Kolory] "
+            } else {
+                "[$sub] "
+            }
+
+            val check = player.getString("url")
+            if (check in listOf("vk", "cda", "mp4upload", "sibnet", "dailymotion")) {
+                val url = getPlayerUrl(id)
+                serverList.add(Pair(url, prefix))
+            }
+        }
+
+        val videoList = serverList.parallelCatchingFlatMap { (serverUrl, prefix) ->
+            when {
+                serverUrl.contains("vk.com") -> {
+                    VkExtractor(client, headers).videosFromUrl(serverUrl, prefix)
```

Comment:
Move out the extractors to a class instance to avoid repeated initialization

Reply:
I will honestly admit that I don't think I understand D:

Reply:
Above the function definition, do something like `private val vkExtractor = VkExtractor(client, headers)` and use `vkExtractor` in the function